From 042d0a1f8cb0cb374aed430a762fa31cbe422442 Mon Sep 17 00:00:00 2001
From: Baptiste Assmann <bedis9@gmail.com>
Date: Wed, 2 Sep 2015 21:52:37 +0200
Subject: [PATCH 10/17] MINOR: dns: new bitmasks to use against DNS flags

Current DNS client code implementation doesn't take care of response
flags setup by the server.
This patch introduces a couple of bitmasks one can use to retrieve the
truncated flag and the reply code available in the 2-bytes flag field.
---
 include/types/dns.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/types/dns.h b/include/types/dns.h
index d0a835d..589bb5d 100644
--- a/include/types/dns.h
+++ b/include/types/dns.h
@@ -51,6 +51,10 @@
 #define DNS_RCODE_NX_DOMAIN	3	/* non existent domain */
 #define DNS_RCODE_REFUSED	5	/* query refused */
 
+/* dns flags masks */
+#define DNS_FLAG_TRUNCATED	0x0200	/* mask for truncated flag */
+#define DNS_FLAG_REPLYCODE	0x000F	/* mask for reply code */
+
 /* DNS request or response header structure */
 struct dns_header {
 	unsigned short	id:16;		/* identifier */
-- 
2.4.6

